projects/ad469x_iio/app/app_config.c

```diff
@@ -0,0 +1,353 @@
+	.port = SPI_CS_PORT_NUM,
+	.number = SPI_CS_PIN_NUM,
+	.platform_ops = &gpio_ops,
+	.extra = &cs_pwm_gpio_extra_init_params
+};
+#endif
+
+/******************************************************************************/
+/************************ Functions Prototypes ********************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/**
+ * @brief 	Initialize the UART peripheral.
+ * @return	0 in case of success, negative error code otherwise.
+ */
+static int32_t init_uart(void)
+{
+	int32_t ret;
+
+	/* Initialize the serial link for IIO communication */
+	ret = no_os_uart_init(&uart_iio_com_desc, &uart_iio_comm_init_params);
+	if (ret) {
+		return ret;
+	}
+
+#if defined(CONSOLE_STDIO_PORT_AVAILABLE)
+	/* Initialize the serial link for console stdio communication */
+	ret = no_os_uart_init(&uart_console_stdio_desc,
+			      &uart_console_stdio_init_params);
+	if (ret) {
+		return ret;
+	}
+#endif
+
+	return 0;
+}
+
+/**
+ * @brief 	Initialize the IRQ controller.
+ * @return	0 in case of success, negative error code otherwise.
+ * @details	This function initialize the interrupts for system peripherals.
+ */
+static int32_t init_interrupt(void)
+{
+	int32_t ret;
+
+	/* Init interrupt controller for external interrupt */
+	ret = no_os_irq_ctrl_init(&trigger_irq_desc, &trigger_gpio_irq_params);
+	if (ret) {
+		return ret;
+	}
+
+#if (DATA_CAPTURE_MODE == BURST_DATA_CAPTURE)
+	/* The BSY pin has been tied as the interrupt source to sense the
+	 * End of Conversion. The registered callback function is responsible
+	 * of reading the raw samples via the SPI bus */
+	ret = no_os_irq_register_callback(trigger_irq_desc,
+					  TRIGGER_INT_ID,
+					  &ext_int_callback_desc);
+	if (ret) {
+		return ret;
+	}
+
+	ret = no_os_irq_trigger_level_set(trigger_irq_desc,
+					  TRIGGER_INT_ID, NO_OS_IRQ_EDGE_FALLING);
+	if (ret) {
+		return ret;
+	}
+
+	ret = no_os_irq_disable(trigger_irq_desc, TRIGGER_INT_ID);
+	if (ret) {
+		return ret;
+	}
+#endif
+	return 0;
+}
+
+/**
+ * @brief 	Initialize the PWM interface.
+ * @return	0 in case of success, negative error code otherwise.
+ */
+int32_t init_pwm(void)
+{
+	int32_t ret;
+
+	/* Initialize the PWM interface to generate conversion signal.
+	 * If SPI_DMA is enabled, pwm interface is also used to
+	 * trigger dummy spi tx dma transcation to fetch MSB of
+	 * sampled data.
+	 * */
+	ret = no_os_pwm_init(&pwm_desc, &pwm_init_params);
+	if (ret) {
+		return ret;
+	}
+
+	ret = no_os_pwm_disable(pwm_desc);
+	if (ret) {
+		return ret;
+	}
+
+#if (INTERFACE_MODE == SPI_DMA)
+	ret = no_os_pwm_init(&tx_trigger_desc, &tx_trigger_init_params);
+	if (ret) {
+		return ret;
+	}
+
+	ret = no_os_pwm_disable(tx_trigger_desc);
```

Comment:
where is tx_trigger_desc enabled?

Reply:
Tx trigger should not be manually enabled, as it is being operated in slave mode, where the CNV timer (timer 1 channel 3) acts as a master.